pkg/ingester/ingester_v2.go

```diff
@@ -1031,6 +1035,56 @@ func (i *Ingester) v2Query(ctx context.Context, req *client.QueryRequest) (*clie
 	return result, ss.Err()
 }
 
+func (i *Ingester) v2QueryExemplars(ctx context.Context, req *client.ExemplarQueryRequest) (*client.ExemplarQueryResponse, error) {
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	from, through, matchers, err := client.FromExemplarQueryRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	i.metrics.queries.Inc()
+
+	db := i.getTSDB(userID)
+	if db == nil {
+		return &client.ExemplarQueryResponse{}, nil
+	}
+
+	// Note that currently Prometheus' exemplar querier does nothing with a context that you pass it.
```

Comment:
[nit] I would remove this comment just because these kind of comments age very badly :) If ctx will be used at some point by Prometheus, we'll very likely forget to remove this comment, but this comment can still deceive the reader/contributor.